homeassistant/components/media_player/androidtv.py

```diff
@@ -0,0 +1,524 @@
+    "left": "21",
+    "menu": "82",
+    "move_home": "122",
+    "mute": "164",
+    "pairing": "225",
+    "power": "26",
+    "resume": "224",
+    "right": "22",
+    "sat": "237",
+    "search": "84",
+    "settings": "176",
+    "sleep": "223",
+    "suspend": "276",
+    "sysdown": "281",
+    "sysleft": "282",
+    "sysright": "283",
+    "sysup": "280",
+    "text": "233",
+    "top": "122",
+    "up": "19",
+    "vga": "251",
+    "voldown": "25",
+    "volup": "24",
+    "yellow": "185"
+}
+
+KNOWN_APPS = {
+    "amazon": "Amazon Prime Video",
+    "dream": "Screensaver",
+    "kodi": "Kodi",
+    "netflix": "Netflix",
+    "plex": "Plex",
+    "spotify": "Spotify",
+    "tvlauncher": "Homescreen",
+    "youtube": "Youtube",
+    "zatto": "Zattoo"
+}
+
+SUPPORT_ANDROIDTV = (SUPPORT_NEXT_TRACK | SUPPORT_PAUSE |
+                     SUPPORT_PLAY | SUPPORT_PREVIOUS_TRACK |
+                     SUPPORT_TURN_OFF | SUPPORT_TURN_ON |
+                     SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP |
+                     SUPPORT_STOP)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.All(cv.port, cv.string),
+    vol.Optional(CONF_ADBKEY): has_adb_files,
+    vol.Optional(CONF_APPS, default=DEFAULT_APPS): dict,
+    vol.Optional(CONF_ADB_SERVER_IP): cv.string,
+    vol.Optional(
+        CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT): cv.port
+})
+
+ACTION_SERVICE = 'androidtv_action'
+INTENT_SERVICE = 'androidtv_intent'
+KEY_SERVICE = 'androidtv_key'
+
+SERVICE_ACTION_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required('action'): vol.In(ACTIONS),
+})
+
+SERVICE_INTENT_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required('intent'): cv.string,
+})
+
+SERVICE_KEY_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required('key'): cv.string,
+})
+
+DATA_KEY = '{}.androidtv'.format(DOMAIN)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the androidtv platform."""
+    from androidtv import AndroidTV
+    if DATA_KEY not in hass.data:
+        hass.data[DATA_KEY] = {}
+
+    host = '{0}:{1}'.format(config[CONF_HOST], config[CONF_PORT])
+    name = config.get(CONF_NAME)
+
+    if CONF_ADB_SERVER_IP not in config:
+        atv = AndroidTV(host)
+        if not atv.available:
+            # "python-adb" with adbkey
+            if CONF_ADBKEY in config:
+                adbkey = config[CONF_ADBKEY]
+            else:
+                adbkey = DEFAULT_ADBKEY
+            atv = AndroidTV(host, adbkey)
+            adb_log = " using adbkey='{0}'".format(adbkey)
+        else:
+            adb_log = ""
+
+    else:
+        # "pure-python-adb"
+        atv = AndroidTV(
+            host,
+            adb_server_ip=config[CONF_ADB_SERVER_IP],
```

Comment:
[Question] Shouldn't you accept the adbkey for remote servers in the case someone is hosting ADB  behind auth?

Reply:
The `adbkey` configuration entry is only needed when using the Python implementation of the ADB protocol (aka, the `adb` package).  When using an ADB server for ADB commands (such as the Hass.io addon), the adb instance is responsible for authenticating.  Since that responsibility is outside the scope of this component, the `adbkey` configuration entry is not relevant.